### PR TITLE
Allow dtype to be specified in Flax pipeline

### PR DESCRIPTION
**Pull request by pcuenca**
This replaces #581, which was reviewed by @patil-suraj.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def __call__(
             self.unet.sample_size,
         )
         if latents is None:
-            latents = jax.random.normal(prng_seed, shape=latents_shape, dtype=self.dtype)
+            latents = jax.random.normal(prng_seed, shape=latents_shape, dtype=jnp.float32)
```
Comment (pcuenca) [inline src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py:160]: The alternative to this, I think, would be to prepare the scheduler parameters using the same `dtype` as the model. We can do that in a follow-up PR.
